File: scala0311/src/main/scala/com/atguigu/excercise/TestMethod.scala
package com.atguigu.excercise

object TestMethod {
  def main(args: Array[String]): Unit = {
    def f1 = "venassa"
    println(f1)

    def test1(x:Double) = {
      (y:Double) => x* x* y
    }

    val res= test1(2.0)

    val res2: Double = res(3.0)

  //  println(res2)

    val a = Array(1,2,3)

    def maxmin(nums:Array[Int])={
      nums.max -> nums.min
    }

    val i: Int = 1
    val ii = 11
    //println(maxmin((1 to 10).toArray))

    val names = List("Petter", "Paul", "Mary")

    val chars: List[Char] = for(name <- names;c <- name) yield c.toUpper

    chars.foreach(println)

  }

}

File: spark0105-0613/src/main/java/com/atguigu/TestMethod.java
package com.atguigu;

public class TestMethod {
    public static void main(String[] args) {
        AA aa = new BB();

        // 动态绑定机制：在调用成员方法时，会将这个方法和对象的实际内存进行绑定,然后调用
        // 属性没有动态绑定机制,在哪里声明在哪使用
        System.out.println(aa.getResult());
    }

}

class AA {
    public int i = 10;

    public int getResult() {
        return getI() + 10;
    }

    private int getI() {
        return i;
    }
}

class BB extends AA {
    public int i = 20;

    //    public int getResult() {
//        return i + 20;
//    }
    public int getI() {
        return i;
    }

    public int getB() {
        return i + 11;
    }
}


